api/queries_pr.go

```diff
@@ -418,8 +428,32 @@ func PullRequestByNumber(client *Client, repo ghrepo.Interface, number int) (*Pu
 				author {
 				  login
 				}
-				commits {
+				commits(last: 1) {
 				  totalCount
+					nodes {
+						commit {
+              oid
+						  statusCheckRollup {
```

Comment:
My slight concern with expanding the `PullRequestByNumber` is that the Checks+Statuses data is unnecessarily fetched every time a user does any operation on a pull request, such as `gh pr view 123 --web`. However, I do understand that this approach is the most feasible to implement, since any other course of action would require us to either duplicate the logic of `shared.PRFromArgs()` to fetch extra fields, or to come up with a shippable solution to https://github.com/cli/cli/pull/1081. I'm only bringing this up as a note that this is a good candidate to revisit in a few weeks 👍